src/map.cpp

```diff
@@ -5415,12 +5415,13 @@ computer *map::computer_at( const tripoint &p )
 
 bool map::point_within_camp( const tripoint &point_check ) const
 {
-    const tripoint omt_check = ms_to_omt_copy( point_check );
-    const point p( omt_check.xy() );
-    for( int x2 = p.x - 2; x2 < p.x + 2; x2++ ) {
-        for( int y2 = p.y - 2; y2 < p.y + 2; y2++ ) {
-            if( cata::optional<basecamp *> bcp = overmap_buffer.find_camp( point( x2, y2 ) ) ) {
-                return ( *bcp )->camp_omt_pos().z == point_check.z;
+    // TODO: fix point types
+    const tripoint_abs_omt omt_check( ms_to_omt_copy( point_check ) );
+    const point_abs_omt p = omt_check.xy();
+    for( int x2 = -2; x2 < 2; x2++ ) {
```

Comment:
Shouldn't this be a `point_range<point_abs_omt> points_in_rectange( point( -2, -2 ), point( 2, 2 ) )`

Reply:
Currently `points_in_rectangle` only supports map-scale points, and `points_in_radius` only supports `tripoint`s so neither is a great fit for this.  Agreed that we need a good solution for such.  But it's out of scope for this PR.